### PR TITLE
Rewrite torchvision packaging

### PR DESCRIPTION
**Pull request by ezyang**
Following a similar line of inquiry to https://github.com/pytorch/audio/pull/217

@peterjc123 I haven't ported the Windows bits to the new infrastructure yet. If I have time later I'll look into it; I'm interested in whether or not we can reasonable do the Windows on CircleCI (so there's only one pipeline we have to look at.)

### CODE REVIEW

**Comment by peterjc123** [comment]
@ezyang That's not possible since CircleCI doesn't support Windows.

**Comment by ezyang** [comment]
@peterjc123 Yep. I should be able to get us expedited though.

**Comment by ezyang** [comment]
@peterjc123 OK, Windows CPU is enabled. Docs are at https://circleci.com/docs/2.0/hello-world-windows/

File: torchvision/datasets/utils.py [hunk 1]
```diff
@@ -83,7 +83,7 @@ def download_url(url, root, filename=None, md5=None):
                 url, fpath,
                 reporthook=gen_bar_updater()
             )
-        except urllib.error.URLError as e:
+        except (urllib.error.URLError, IOError) as e:
```
Comment (fmassa) [inline torchvision/datasets/utils.py]: Can you explain why this change was needed?

From https://stackoverflow.com/questions/29347790/difference-between-ioerror-and-oserror, `IOError` is (in Python3) an alias for `OSError`, and [`URLError`](https://docs.python.org/3/library/urllib.error.html#urllib.error.URLError) is a subclass of `IOError` in Py2 and of `OSError` in Python3.

File: test/test_cpp_models.py [hunk 2]
```diff
@@ -66,56 +71,72 @@ def test_vgg16_bn(self):
     def test_vgg19_bn(self):
         process_model(models.vgg19_bn(self.pretrained), self.image, _C_tests.forward_vgg19bn, 'VGG19BN')
 
+    @unittest.expectedFailure
```
Comment (fmassa) [inline test/test_cpp_models.py]: Wait, are all those tests failing for you on Linux? I'm ok disabling those, just wanted to understand

File: torchvision/datasets/utils.py (hunk 1 above)
Comment (ezyang) [reply torchvision/datasets/utils.py]: On Python 2.7 I observed that a certificate validation failure results in an IOError, not a URLError:

```

    def do_handshake(self, block=False):
        """Perform a TLS/SSL handshake."""
        self._check_connected()
        timeout = self.gettimeout()
        try:
            if timeout == 0.0 and block:
                self.settimeout(None)
>           self._sslobj.do_handshake()
E           IOError: [Errno socket error] [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:727)
```

File: test/test_cpp_models.py (hunk 2 above)
Comment (ezyang) [reply test/test_cpp_models.py]: Yes, they fail for me on Linux.

File: torchvision/datasets/utils.py (hunk 1 above)
Comment (fmassa) [reply torchvision/datasets/utils.py]: But from https://docs.python.org/3/library/urllib.error.html#urllib.error.URLError, `URLError` is a subclass of `IOError` in Python 2 . Was it failing?

File: test/test_cpp_models.py (hunk 2 above)
Comment (fmassa) [reply test/test_cpp_models.py]: cc @ShahriarSS we are disabling those tests for now

File: torchvision/datasets/utils.py (hunk 1 above)
Comment (fmassa) [reply torchvision/datasets/utils.py]: Oh, I see, forget what I said.

**Comment by ezyang** [comment]
@peterjc123 I'm leaving some of the old files around so maybe the Windows pipeline keeps working; I don't know how to test it, if you let me know I can try and fix it as well for the file changes

File: test/test_cpp_models.py (hunk 2 above)
Comment (ShahriarSS) [reply test/test_cpp_models.py]: @ezyang Why are the tests failing? We should fix them. Can you give me an output?
